### ISSUE TITLE
the codegen Jsonadapter will casue non - null  JsonDataException

### ISSUE THREAD

**Issue by changety** [square/moshi#1011]
this is my data class
```kotlin
@JsonClass(generateAdapter = true)
data class OpusUrls (
    val playMusicSize: Int,
    val playUrl: String,
    val userRawVideoList: List<UserVideoUrl>? = null
)
```

this is the response : 

```json
{
	"code": 200,
	"message": "xxx",
	"data": {
		"playUrl": "xxx",
		"playMusicSize": xxx,
		"userRawVideoList": null ,  here is the point
	}
}
```
and this is the exception:
```
**com.squareup.moshi.JsonDataException: Non-null value 'userRawVideoList' was null at $.data.userRawVideoList**

2019-11-08 10:50:22.738 1864-2017/com.xxx W/System.err: com.squareup.moshi.JsonDataException: Non-null value 'userRawVideoList' was null at $.data.userRawVideoList
2019-11-08 10:50:22.738 1864-2017/com.xxx W/System.err:     at com.squareup.moshi.internal.Util.unexpectedNull(Util.java:621)
2019-11-08 10:50:22.738 1864-2017/com.xxx W/System.err:     at com.xxx.OpusUrlsJsonAdapter.fromJson(OpusUrlsJsonAdapter.kt:62)
2019-11-08 10:50:22.738 1864-2017/com.xxx W/System.err:     at com.xxx.OpusUrlsJsonAdapter.fromJson(OpusUrlsJsonAdapter.kt:20)
2019-11-08 10:50:22.738 1864-2017/com.xxx W/System.err:     at com.squareup.moshi.internal.NullSafeJsonAdapter.fromJson(NullSafeJsonAdapter.java:40)
2019-11-08 10:50:22.738 1864-2017/com.xxx W/System.err:     at com.squareup.moshi.kotlin.reflect.KotlinJsonAdapter.fromJson(KotlinJsonAdapter.kt:84)
2019-11-08 10:50:22.738 1864-2017/com.xxx W/System.err:     at com.squareup.moshi.internal.NullSafeJsonAdapter.fromJson(NullSafeJsonAdapter.java:40)
```
this is not happen when not use codegen,   I think codegen adapter not understand the nullable quailfer "?"
the code in codegen adapter is here 

```kotlin
4 -> {
          userRawVideoList = listOfUserVideoUrlAdapter.fromJson(reader) ?:
              throw Util.unexpectedNull("userRawVideoList", "userRawVideoList", reader)
// in this line ,  will throw the exception， when the response value is null
          // $mask = $mask and (1 shl 4).inv()
          mask0 = mask0 and 0xffffffef.toInt()
        }
```

**Comment by ZacSweers** [comment]
This issue is fixed on master and a dupe of #990

**Comment by changety** [comment]
copy,   thank you @ZacSweers
